processing/src/main/java/org/apache/druid/segment/DictionaryEncodedColumnMerger.java

```diff
@@ -384,7 +387,10 @@ public void writeIndexes(@Nullable List<IntBuffer> segmentRowNumConversions) thr
     }
   }
 
-
+  protected DictionaryWriter<T> getWriter(String fileName)
```

Comment:
I did a default implementation so that this change is less disruptive, i initially had it as abstract. The issue is that the writer and other stuff are not created/open at the time the merger is constructed, rather it happens when the merging starts. Changing that sounds a bit disruptive and I'd rather not make that change as part of this PR, but am happy to change this method back to being abstract